index.d.ts

```diff
@@ -0,0 +1,131 @@
+export declare function removeListener(event: "scanStop", listener: () => void): events.EventEmitter;
+export declare function removeListener(event: "discover", listener: (peripheral: Peripheral) => void): events.EventEmitter;
+export declare function removeListener(event: string, listener: Function): events.EventEmitter;
+
+export declare function removeAllListeners(event?: string): events.EventEmitter;
+
+export declare var state:string;
+
+export declare class Peripheral extends events.EventEmitter {
+    id:            string;
+    uuid:          string;
+    address:       string;
+    addressType:   string;
+    connectable:   boolean;
+    advertisement: Advertisement;
+    rssi:          number;
+    services:      Service[];
+    state:         'error' | 'connecting' | 'connected' | 'disconnecting' | 'disconnected';
+
+    connect(callback?: (error: string) => void): void;
+    disconnect(callback?: () => void): void;
+    updateRssi(callback?: (error: string, rssi: number) => void): void;
+    discoverServices(serviceUUIDs: string[], callback?: (error: string, services: Service[]) => void): void;
+    discoverAllServicesAndCharacteristics(callback?: (error: string, services: Service[], characteristics: Characteristic[]) => void): void;
+    discoverSomeServicesAndCharacteristics(serviceUUIDs: string[], characteristicUUIDs: string[], callback?: (error: string, services: Service[], characteristics: Characteristic[]) => void): void;
+
+    readHandle(handle: Buffer, callback: (error: string, data: Buffer) => void): void;
+    writeHandle(handle: Buffer, data: Buffer, withoutResponse: boolean, callback: (error: string) => void): void;
+    toString(): string;
+
+    on(event: "connect", listener: (error: string) => void): this;
+    on(event: "disconnect", listener: (error: string) => void): this;
+    on(event: "rssiUpdate", listener: (rssi: number) => void): this;
+	on(event: "servicesDiscover", listener: (services: Service[]) => void): this;
+	on(event: string, listener: Function): this;
+}
+
+export interface Advertisement {
+    localName: string;
+    serviceData: Array<{
+        uuid: string,
+        data: Buffer
+    }>;
+    txPowerLevel: number;
+    manufacturerData: Buffer;
+    serviceUuids: string[];
+}
+
+export declare class Service extends events.EventEmitter {
+    uuid: string;
+    name: string;
+    type: string;
+    includedServiceUuids: string[];
+    characteristics: Characteristic[];
+
+    discoverIncludedServices(serviceUUIDs: string[], callback?: (error: string, includedServiceUuids: string[]) => void): void;
+    discoverCharacteristics(characteristicUUIDs: string[], callback?: (error: string, characteristics: Characteristic[]) => void): void;
+    toString(): string;
+
+    on(event: "includedServicesDiscover", listener: (includedServiceUuids: string[]) => void): this;
+	on(event: "characteristicsDiscover", listener: (characteristics: Characteristic[]) => void): this;
```

Comment:
is indenting correct ? if not please add an other patch to check formating using eslint

Reply:
It was copied directly from DefinitelyTyped ;), have updated formatting